lib/uploader.js

```diff
@@ -355,11 +355,15 @@ function parseResult(buffer, res) {
   return result;
 }
 
-function call_api(action, callback, options, get_params) {
-  if (typeof callback !== "function") {
-    callback = function () {};
+function handleError(callback, result, deferred, error) {
```

Comment:
There are a couple of these which are identical.

I think we should either rename to make them distinct or we can try to share them.
Which do you think makes more sense?

Reply:
updated